### ISSUE TITLE
Bug: Wrong type names in generated code for some entity names

### ISSUE THREAD

**Issue by NexZhu** [facebookincubator/ent#652]
Tested entc and ent version: v0.2.7

Steps to reproduce:
```shell
entc init Vm
go generate ./ent
```

Here's examples of problematic generated code:

In `vm_create.go`:
```go
// VMCreate is the builder for creating a Vm entity.
type VMCreate struct {
	config
	mutation *VMMutation
	hooks    []Hook
}
```

In `client.go`:
```go
// Create returns a create builder for Vm.
func (c *VmClient) Create() *VmCreate {
	mutation := newVMMutation(c.config, OpCreate)
	return &VmCreate{config: c.config, hooks: c.Hooks(), mutation: mutation}
}
```

One is `VMCreate`, the other one is `VmCreate`.

**Comment by NexZhu** [comment]
I glance through the code and the problem is:

The tempaltes for `VmCreate`, `VmDelete` etc., use the `pascal` function in `ent/entc/gen/func.go`, [link](https://github.com/facebookincubator/ent/blob/dd3cdc147dec9b76149ecec80570bfc676650a4f/entc/gen/func.go#L117:6), which check for common uppercase acronyms, while the template for `client.go` doesn't.

```
// Create returns a create builder for {{ $n.Name }}.
func (c *{{ $client }}) Create() *{{ $n.Name }}Create {
	mutation := new{{ $n.MutationName }}(c.config, OpCreate)
	return &{{ $n.Name }}Create{config: c.config, hooks: c.Hooks(), mutation: mutation}
}
```

Should be pretty simple to fix, maybe use a `$n.PascalName` field in the above template.

**Comment by a8m** [comment]
Hey @NexZhu and thanks for reporting this issue.
From quick look at this, it looks like it happens because the schema name ("Vm") is defined in the common acronyms (as "VM") and your schema named "Vm".

Can you please try to use the name "VM" and update if it solved the problem?

**Comment by NexZhu** [comment]
> Hey @NexZhu and thanks for reporting this issue.
> From a quick look at this, it looks like it happens because the schema name ("Vm") is defined in the common acronyms (as "VM") and your schema named "Vm".
> 
> Can you please try to use the name "VM" and update if it solved the problem?

Yes, I saw that code (about common acronyms) earlier and I tried with "VM" which solved the problem. It's still a bug though, so I'm just reporting to let you know.

**Comment by NexZhu** [comment]
From a quick search, it seems that this naming problem exists in quite a lot of places.

I did a quick search and edit, and submitted a pull request for reference:
https://github.com/facebookincubator/ent/pull/654

You can see the related places in code.

**Comment by a8m** [comment]
Thanks for the reporting @NexZhu.
Fixed with #653
